src/Service/TeamService.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace App\Service;
+
+use App\Controller\Traits\ApiResponseTrait;
+use App\Entity\League;
+use App\Entity\Team;
+use App\Repository\LeagueRepository;
+use App\Repository\TeamRepository;
+
+use Doctrine\DBAL\Exception\UniqueConstraintViolationException;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class TeamService
+{
+    use ApiResponseTrait {
+        getRequestParams as _getRequestParams;
+    }
+
+    private $em = null;
+
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
+    }
+
+    /**
+     * @param Request $request
+     * @return bool|\Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function validateCreate(Request $request)
+    {
+        $this->getRequestParams($request);
+
+        // check required fields
+        if (empty($this->requestParams['name'])) {
+            $this->errorFields['name'] = ApiCodes::ERR_REQUIRED_PARAM;
+        }
+        if (empty($this->requestParams['strip'])) {
+            $this->errorFields['strip'] = ApiCodes::ERR_REQUIRED_PARAM;
+        }
+        if (empty($this->requestParams['league_id'])) {
+            $this->errorFields['league_id'] = ApiCodes::ERR_REQUIRED_PARAM;
+        }
+
+        if (!empty($this->errorFields)) {
+            return $this->responseJson([], Response::HTTP_BAD_REQUEST);
+        }
+        return true;
+    }
+
+    /**
+     * @param Request $request
+     * @param bool $returnCount
+     * @return mixed
+     */
+    public function getPaginationList(Request $request, $returnCount = false)
```

Comment:
In general, using the true/false flags it's not a right way, match better - have a couple of public methods, that will have only one responsibility, common code must be moved to protected/private methods